### PR TITLE
Add audit event support to client CLI #272

### LINKED ISSUE #272: Add audit events command to interactive client
The interactive client doesn't currently support retrieving audit events even though the client does. This command should be added.

- [x] `audit` should retrieve all audit events with no filters
- [x] `audit type:["all" | "system-only" | "key-only"]` should specify the given event type
- [x] ~`audit before:[yyyy-mm-dd]` should use a before-date filter with midnight as the time~
- [x] ~`audit after:[yyyy-mm-dd]` should use an after-date filter with midnight as the time~
- [x] `audit keys:[key_name_1, key_name_2, ...]` should use a key_id filter. Keys should be specified by name and key_ids looked up by `lkic`
- [x] `audit request-id:[uuid]`
- [x] All of the above options should be combinable
- [x] Replace existing `println!("Not implemented")` block with a proper command
- [x] Add documentation to the `HELP_TEXT` constant and README
- [x] Add example to README

### PR DESCRIPTION
**Pull request by picklenerd**
Client CLI can now retrieve audit event logs. I implemented all of the filters except for the date filters since they depend on `mongodb` types which will be replaced soon.

Other changes:
- Removed help text from README. This will end up being outdated so I don't think it's worth trying to maintain.
- Remote signing now accepts all remaining text after the key name as the data to be signed. This lets you use spaces in your data string.

I added a `merge` method to `AuditEventOptions` which didn't end up being used due to the `CliComand` trait definition for `parse_command_args`. I still plan to use this method after more refactoring so I'd like to leave this in.

Closes #272

### CODE REVIEW

File: bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs [hunk 1]
```diff
@@ -1,29 +1,117 @@
...
+        for event in audit_event_log {
+            println!("----------------------------------");
+            println!("Request ID: {}", event.request_id());
+            if let Some(key_id) = event.key_id() {
+                println!("{key_id:?}");
+            }
+            println!("{}", event.date());
+            println!("{}", event.action());
+            println!("{}", event.status());
+        }
```
Comment (vidyaap) [inline bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs:69]: How do you feel about using this format for the Display implementation of AuditEvent, and then printing the event here directly? Right now, its Display impl is a full sentence, but I think this is easier to read/parse.

File: bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs [hunk 2]
```diff
@@ -37,6 +125,45 @@ impl CliCommand for GetAuditEvents {
...
+            let key_names: Vec<String> = option_value.split(',').map(ToString::to_string).collect();
+            Ok(ParsedAuditEventOption::KeyNames(key_names))
+        }
+        "request-id" => Ok(ParsedAuditEventOption::RequestId(option_value.parse()?)),
+        "before" => {
+            anyhow::bail!("`before` option is not implemented");
+        }
+        "after" => {
+            anyhow::bail!("`after` option is not implemented");
+        }
```
Comment (vidyaap) [inline bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs:160]: I saw the date filters weren't checked off on the issue either - is the plan to do those in another PR or to not include them in the CLI?

File: lock-keeper/src/types/audit_event.rs [hunk 3]
```diff
@@ -110,10 +111,164 @@ impl EventType {
...
+            match &mut self.key_ids {
+                Some(key_ids) => {
+                    key_ids.extend(incoming_key_ids);
+                }
+                None => self.key_ids = Some(incoming_key_ids),
+            }
+        }
+
+        self.after_date = other.after_date.or(self.after_date);
+        self.before_date = other.before_date.or(self.before_date);
```
Comment (vidyaap) [inline lock-keeper/src/types/audit_event.rs:138]: I'm not sure I understand what's happening with the dates when you merge. The tests make it seem like the original dates get overwritten with the merged dates, but I'm not sure what that means semantically when you merge options? I think I would have expected to pick the "earlier after" and the "later before" if that makes sense.

File: lock-keeper/src/types/audit_event.rs [hunk 4]
```diff
@@ -83,7 +83,8 @@ impl Display for AuditEvent {
 }
 
 /// Options for which types of events to retrieve from the key server
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, EnumString, Display)]
+#[strum(serialize_all = "kebab-case")]
 pub enum EventType {
```
Comment (vidyaap) [inline lock-keeper/src/types/audit_event.rs:88]: Could you add GetUserId to system-only events? I'm not sure it should be a key event.

File: lock-keeper/src/types/audit_event.rs (hunk 3 above)
Comment (picklenerd) [reply lock-keeper/src/types/audit_event.rs:138]: Yeah it currently overwrites but I like your idea better!

Comment (picklenerd) [reply lock-keeper/src/types/audit_event.rs:138]: I don't want to write a bunch of date stuff since the types are about to change so I think I'm just going to throw this merge function in a branch and remove it from this PR.

File: bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs (hunk 2 above)
Comment (picklenerd) [reply bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs:160]: I'll make a new ticket for those.

File: bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs (hunk 1 above)
Comment (picklenerd) [reply bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs:69]: I can get behind that.

File: lock-keeper/src/types/audit_event.rs (hunk 4 above)
Comment (picklenerd) [reply lock-keeper/src/types/audit_event.rs:90]: Added. I changed the way the filters work a bit so that they use a constant instead of allocating new vectors.

File: bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs (hunk 2 above)
Comment (picklenerd) [reply bin/lock-keeper-client-cli/src/cli_command/get_audit_events.rs:154]: #434

**Review by vidyaap — APPROVED** [review]
Thanks, looks good to me!
